loader_hub/confluence/base.py

```diff
@@ -0,0 +1,92 @@
+"""Confluence reader."""
+import os
+from typing import List, Optional
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+CONFLUENCE_USERNAME = "CONFLUENCE_USERNAME"
+CONFLUENCE_API_TOKEN = "CONFLUENCE_API_TOKEN"
+
+
+class ConfluenceReader(BaseReader):
+    """Confluence reader.
+
+    Reads a set of confluence pages given a space id and optionally a list of page ids
+
+    Args:
+        user_name (str): your confluence username.
+        api_token (str): api token for your account (https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/)
+        base_url (str): 'base_url' for confluence instance, this is suffixed with '/wiki', eg 'https://yoursite.atlassian.com/wiki'
+
+    """
+
+    def __init__(self, user_name: Optional[str] = None, api_token: Optional[str] = None, base_url: str = None) -> None:
+
+        try:
+            from atlassian import Confluence
+        except ImportError:
+            raise ImportError("`atlassian` package not found, please run `pip install atlassian-python-api`")
+
+        if user_name is None:
+            user_name = os.getenv(CONFLUENCE_USERNAME)
+            if user_name is None:
+                raise ValueError(
+                    "Must specify `user_name` or set environment "
+                    "variable `CONFLUENCE_USERNAME`."
+                )
+        if api_token is None:
+            api_token = os.getenv(CONFLUENCE_API_TOKEN)
+            if api_token is None:
+                raise ValueError(
+                    "Must specify `api_token` or set environment "
+                    "variable `CONFLUENCE_API_TOKEN`."
+                )
+
+        self.confluence = Confluence(url=base_url, username=user_name, password=api_token, cloud=True)
+
+    def load_data(self, space_id: Optional[str] = None, page_ids: Optional[List[str]] = []) -> List[Document]:
```

Comment:
nit: I'd set default to `page_ids` to None, and do `page_ids = page_ids or []` within the function. There's some dangers to initializing an empty list in the function signature